src/EditorFeatures/Core/Shared/Preview/PreviewSolutionCrawlerRegistrationService.cs

```diff
@@ -101,7 +101,12 @@ private async Task AnalyzeAsync()
                 }
             }
 
-            public async void Unregister(Workspace workspace, bool blockingShutdown = false)
+            public void Unregister(Workspace workspace, bool blockingShutdown = false)
+            {
+                _ = UnregisterAsync(workspace, blockingShutdown);
+            }
+
+            private async Task UnregisterAsync(Workspace workspace, bool blockingShutdown)
             {
```

Comment:
based on previous comments, this actually feels like a bug.  This code seems to be missing required try/catches, and so could tear things down if something goes wrong.

Reply:
➡️ Covered under the VSTHRD110 (fire and forget) umbrella